pressure_traction_test for Fluent v2020R1 fails on cfdclu55

`FAIL: test_pressure_traction (coconut.tests.solver_wrappers.fluent.test_v2020R1.TestSolverWrapperFluent2020R1Tube3D)`
----------------------------------------------------------------------
```
Traceback (most recent call last):
  File "/cfdfile2/data/fm/nicolas/python_packages/coconut/coconut/tests/solver_wrappers/fluent/test_v2019R1.py", line 314, in test_pressure_traction
    np.testing.assert_allclose(traction[0], traction[2], rtol=1e-11)
  File "/apps/SL6.3/Anaconda/2019.07/python3.7/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 1501, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/apps/SL6.3/Anaconda/2019.07/python3.7/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 827, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-11, atol=0

 

Mismatch: 0.0217%
Max absolute difference: 3.33955086e-13
Max relative difference: 1.2415593e-11
 x: array([[-10.349147,  -8.341815,  -4.579155],
       [ 46.352385,   2.219764,   2.705838],
       [ 26.331443,  -7.791217,  -2.45801 ],...
 y: array([[-10.349147,  -8.341815,  -4.579155],
       [ 46.352385,   2.219764,   2.705838],
       [ 26.331443,  -7.791217,  -2.45801 ],...
```
----------------------------------------------------------------------


Test uses maximum number of cores on machine (in this case 24 cores).
For other Fluent versions the test does work.
Either tolerance should be lowered or number of cores limited.

--- a/tests/solver_wrappers/fluent/test_v2019R1.py
+++ b/tests/solver_wrappers/fluent/test_v2019R1.py
@@ -92,7 +92,7 @@ class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
         # adapt parameters, create solver
 
         # fix cores based on available cores
-        max_cores = multiprocessing.cpu_count() # max available cores
+        max_cores = multiprocessing.cpu_count()  # max available cores
         if max_cores >= 8:
             cores = 8
         elif max_cores >= 4:
@@ -280,7 +280,7 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         # adapt parameters, create solver
 
         # fix cores based on available cores
-        max_cores = multiprocessing.cpu_count() #max available cores
+        max_cores = multiprocessing.cpu_count()  # max available cores
         if max_cores >= 8:
             cores = 8
         elif max_cores >= 4:

--- a/tests/solver_wrappers/fluent/test_v2019R1.py
+++ b/tests/solver_wrappers/fluent/test_v2019R1.py
@@ -8,31 +8,10 @@ import subprocess
 import json
 import multiprocessing
 
-
-
-
 # TODO: issue: id is a Python build-in function... use ids instead?
 
-# TODO: put somewhere in docs?
-"""
-- The unittests in this file (i.e. for v2019R1) can be run
-directly, using
-
-    python -m unittest -vb solver_wrappers/fluent/test_v2019R1.py
-
-- The unittests for the other Fluent versions currently 
-inherit from these classes. Because those modules import
-these super classes, they are also run and crash, which is
-kinda ugly. 
-To run other Fluent version unittests, adapt the file
-__ini__.py and use
-
-    python -m unittest discover -vb
-    
-- The unittests for different Fluent versions can never be
-run simultaneously, because the correct software must be 
-loaded in advance. 
-"""
+# TODO: The unittests for the other Fluent versions currently inherit from these classes. Because those modules import
+#   these super classes, they are also run and crash, which is kinda ugly.
 
 
 class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
@@ -53,6 +32,7 @@ class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
         self.mp_name_in = 'beamoutside_nodes'
         self.mp_name_out = 'beamoutside_faces'
 
+    # noinspection PyMethodMayBeStatic
     def get_dy(self, x):
         return 0.0001 * np.sin(2 * np.pi / 0.05 * x)
 
@@ -115,9 +95,9 @@ class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
         max_cores = multiprocessing.cpu_count() # max available cores
         if max_cores >= 8:
             cores = 8
-        elif 8 > max_cores >= 2:
+        elif max_cores >= 4:
             cores = 4
-        elif 4 > max_cores >= 2:
+        elif max_cores >= 2:
             cores = 2
         else:
             cores = 1
@@ -233,6 +213,7 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         self.mp_name_in = 'wall_nodes'
         self.mp_name_out = 'wall_faces'
 
+    # noinspection PyMethodMayBeStatic
     def get_dy_dz(self, x, y, z):
         dr = 0.0001 * np.sin(2 * np.pi / 0.05 * x)
         theta = np.arctan2(z, y)
@@ -302,9 +283,9 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         max_cores = multiprocessing.cpu_count() #max available cores
         if max_cores >= 8:
             cores = 8
-        elif 8 > max_cores >= 2:
+        elif max_cores >= 4:
             cores = 4
-        elif 4 > max_cores >= 2:
+        elif max_cores >= 2:
             cores = 2
         else:
             cores = 1

--- a/tests/solver_wrappers/fluent/test_v2019R1.py
+++ b/tests/solver_wrappers/fluent/test_v2019R1.py
@@ -6,6 +6,10 @@ import os
 from os.path import join
 import subprocess
 import json
+import multiprocessing
+
+
+
 
 # TODO: issue: id is a Python build-in function... use ids instead?
 
@@ -106,7 +110,17 @@ class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
         # test if same coordinates always give same pressure & traction
 
         # adapt parameters, create solver
-        self.parameters['settings']['cores'] = 0
+        ## fix cores based on available cores
+        max_cores = multiprocessing.cpu_count()
+        if max_cores >= 8:
+            cores = 8
+        elif 8 > max_cores >= 2:
+            cores = 4
+        elif 4 > max_cores >= 2:
+            cores = 2
+        else:
+            cores = 1
+        self.parameters['settings']['cores'] = cores
         self.parameters['settings']['flow_iterations'] = 500
         solver = create_instance(self.parameters)
         solver.initialize()
@@ -207,7 +221,7 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         if cls.setup_case:
-            dir_tmp = join(os.getcwd(),f'solver_wrappers/fluent/test_v{cls.version}/tube3d')
+            dir_tmp = join(os.getcwd(), f'solver_wrappers/fluent/test_v{cls.version}/tube3d')
             p = subprocess.Popen(join(dir_tmp, 'setup_fluent.sh'), cwd=dir_tmp, shell=True)
             p.wait()
 
@@ -282,7 +296,17 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         # test if same coordinates always give same pressure & traction
 
         # adapt parameters, create solver
-        self.parameters['settings']['cores'] = 0
+        ## fix cores based on available cores
+        max_cores = multiprocessing.cpu_count()
+        if max_cores >= 8:
+            cores = 8
+        elif 8 > max_cores >= 2:
+            cores = 4
+        elif 4 > max_cores >= 2:
+            cores = 2
+        else:
+            cores = 1
+        self.parameters['settings']['cores'] = cores
         self.parameters['settings']['flow_iterations'] = 500
         solver = create_instance(self.parameters)
         solver.initialize()

--- a/tests/solver_wrappers/fluent/test_v2019R1.py
+++ b/tests/solver_wrappers/fluent/test_v2019R1.py
@@ -310,8 +310,8 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         solver.finalize()
 
         # check if same position gives same pressure & traction
-        np.testing.assert_allclose(pressure[0], pressure[2], rtol=1e-11)
-        np.testing.assert_allclose(traction[0], traction[2], rtol=1e-11)
+        np.testing.assert_allclose(pressure[0], pressure[2], rtol=1e-10)
+        np.testing.assert_allclose(traction[0], traction[2], rtol=1e-10)
 
         # check if different position gives different pressure & traction
         p01 = np.linalg.norm(pressure[0] - pressure[1])

--- a/tests/solver_wrappers/fluent/test_v2019R1.py
+++ b/tests/solver_wrappers/fluent/test_v2019R1.py
@@ -110,8 +110,9 @@ class TestSolverWrapperFluent2019R1Tube2D(unittest.TestCase):
         # test if same coordinates always give same pressure & traction
 
         # adapt parameters, create solver
-        ## fix cores based on available cores
-        max_cores = multiprocessing.cpu_count()
+
+        # fix cores based on available cores
+        max_cores = multiprocessing.cpu_count() # max available cores
         if max_cores >= 8:
             cores = 8
         elif 8 > max_cores >= 2:
@@ -296,8 +297,9 @@ class TestSolverWrapperFluent2019R1Tube3D(unittest.TestCase):
         # test if same coordinates always give same pressure & traction
 
         # adapt parameters, create solver
-        ## fix cores based on available cores
-        max_cores = multiprocessing.cpu_count()
+
+        # fix cores based on available cores
+        max_cores = multiprocessing.cpu_count() #max available cores
         if max_cores >= 8:
             cores = 8
         elif 8 > max_cores >= 2:

